gluten-cbo/common/src/main/scala/io/glutenproject/cbo/CboCluster.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.cbo
+
+import io.glutenproject.cbo.memo.MemoTable
+
+import scala.collection.mutable
+
+trait CboClusterKey
+
+object CboClusterKey {
+  implicit class CboClusterKeyImplicits[T <: AnyRef](key: CboClusterKey) {
+    def propSets(memoTable: MemoTable[T]): Set[PropertySet[T]] = {
+      memoTable.getClusterPropSets(key)
+    }
+  }
+}
+
+trait CboCluster[T <: AnyRef] {
```

Comment:
`CboCluster` is a set of nodes sharing the same context in the original input plan. One cluster can derives a subset of `CboGroup`s. Nodes in one `CboGroup` shares the same properties.

Reply:
Add commets?

Reply:
Will raise a independent PR to add code doc. The code is under frequent modification as of now.